### ISSUE TITLE
runtime: GODEBUG gctrace bug with time ran on Mac

### ISSUE THREAD

**Issue by ardan-bkennedy** [golang/go#21554]
### What version of Go are you using (`go version`)?
go version go1.9rc2 darwin/amd64

### Does this issue reproduce with the latest release?
Yes, only in 1.9 and since beta.

### What operating system and processor architecture are you using (`go env`)?
```
GOARCH="amd64"
GOHOSTARCH="amd64"
GOHOSTOS="darwin"
GOOS="darwin"

macOS Sierra  Version 10.12.6
```

### What did you do?

Using this program:
https://github.com/ardanlabs/gotraining/tree/master/topics/go/profiling/project

Build the code and run it using:
```$ GODEBUG=gctrace=1 ./project > /dev/null```

Then open the browser and run a search. Be sure to check the three boxes for cnn, bbc and nyt and you get some results.
http://localhost:5000/search

This will produce a gctrace immediately.

### What did you expect to see?

gc 1 **@1.186s** 0%: 0.092+0.47+0.038 ms clock, 0.37+0.15/0.81/1.8+0.15 ms cpu, 4->4->1 MB, 5 MB goal, 8 P

### What did you see instead?

gc 1 **@18446653480.186s** 0%: 0.092+0.47+0.038 ms clock, 0.37+0.15/0.81/1.8+0.15 ms cpu, 4->4->1 MB, 5 MB goal, 8 P

**Comment by ALTree** [comment]
cc @aclements

**Comment by aclements** [comment]
It took me some work, but I was able to reproduce this on a gomote

```
git checkout go1.9rc2
VM=$(gomote create darwin-amd64-10_12)
gomote push $VM
gomote ssh $VM
cd $GOROOT/src
./make.bash
cd ../bin
PATH=$PWD:$PATH
go get -d github.com/ardanlabs/gotraining/topics/go/profiling/project
cd $GOPATH/src/github.com/ardanlabs/gotraining/topics/go/profiling/project
go build
GODEBUG=gctrace=1 ./project > /dev/null
```

In another terminal:
```
gomote ssh $VM
curl -d cnn=on -d nyt=on -d bbc=on -d term=test http://localhost:5000/search
```

**Comment by aclements** [comment]
Ah, the problem is we have two different time bases in the runtime as of e4371fb179ad69cbd057f2430120843948e09f2f. That commit made `nanotime()` return a value relative to `runtime.startNano`. However, `runtime.main` does `runtimeInitTime = nanotime()` *before* `runtime_init()`, which sets `startNano = nanotime()`. The gctrace times are computed as `nanotime() - runtimeInitTime`, so when the time base changes, this no longer makes sense.

**Comment by aclements** [comment]
Re-opening for cherry-pick to Go 1.9.1

**Comment by ianlancetaylor** [comment]
Is this worth cherry picking?  It seems a very minor problem.

**Comment by aclements** [comment]
Rick and I pretty frequently ask people for gctrace output when debugging performance issues. This could make that harder to interpret. It's also a simple fix, so I think whenever 1.9.1 happens we might as well include it.

**Comment by ardan-bkennedy** [comment]
I am teaching this stuff and now that 1.9 has been released, this bug makes things difficult for training. So please consider getting this in sooner rather then later.

**Comment by rsc** [comment]
OK for Go 1.9.2.

**Comment by rsc** [comment]
go1.9.2 has been packaged and includes:

  - [CL 58690](https://golang.org/cl/58690) 

The release is posted at [golang.org/dl](https://golang.org/dl).

— golang.org/x/build/cmd/releasebot, Oct 26 21:09:07 UTC

**Comment by ardan-bkennedy** [comment]
Just tried it out on darwin and it looks good!  Thank You.

Is there anyway this can get fixed as well for 1.9? It is broken on both darwin and linux.
https://github.com/golang/go/issues/21697

**Comment by aclements** [comment]
@ardan-bkennedy, this isn't really the place to ask about the other issue. If you're interested in having #21697 fixed in 1.9, you should argue that on that issue (and understand that whether or not the fix gets backported also depends on the risk involved in the actual fix).

**Comment by ardan-bkennedy** [comment]
New to all this so thanks for laying out the policy I am supposed to follow. It won’t happen again.
